core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -12721,6 +12735,74 @@ public final List<User> getUserListOfRoleWithID(String roleName, String filter,
         return users;
     }
 
+    @Override
+    public final List<User> getUserListOfGroupWithID(String groupName) throws UserStoreException {
+
+        if (!isSecureCall.get()) {
+            Class argTypes[] = new Class[]{String.class};
+            Object object = callSecure("getUserListOfGroupWithID", new Object[]{groupName}, argTypes);
+            return (List<User>) object;
+        }
+
+        return getUserListOfGroupWithID(groupName, QUERY_FILTER_STRING_ANY, QUERY_MAX_ITEM_LIMIT_ANY);
+    }
+
+    @Override
+    public final List<User> getUserListOfGroupWithID(String groupName, String filter, int maxItemLimit)
+            throws UserStoreException {
+
+        if (!isSecureCall.get()) {
+            Class argTypes[] = new Class[]{String.class, String.class, int.class};
+            Object object = callSecure("getUserListOfGroupWithID", new Object[]{groupName, filter, maxItemLimit},
+                    argTypes);
+            return (List<User>) object;
+        }
+
+        List<User> users = new ArrayList<>();
+
+        // If group does not exit, just return
+        if (!isExistingRole(groupName)) {
+            handleDoPostGetUserListOfRoleWithID(groupName, users);
+            return users;
+        }
+
+        UserStore userStore = getUserStoreOfRoles(groupName);
+
+        if (userStore.isRecurssive()) {
+            UserStoreManager resolvedUserStoreManager = userStore.getUserStoreManager();
+            if (resolvedUserStoreManager instanceof AbstractUserStoreManager) {
+                return ((AbstractUserStoreManager) resolvedUserStoreManager)
+                        .getUserListOfGroupWithID(userStore.getDomainFreeName(), filter, maxItemLimit);
+            } else {
+                return ((UniqueIDUserStoreManager) resolvedUserStoreManager)
+                        .getUserListOfGroupWithID(userStore.getDomainFreeName());
+            }
```

Comment:
without adding else {}

we could add  return ((UniqueIDUserStoreManager) resolvedUserStoreManager)
                        .getUserListOfGroupWithID(userStore.getDomainFreeName())

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/4003